src/wp-includes/abilities-api/class-wp-abilities-registry.php

```diff
@@ -78,11 +78,11 @@ final class WP_Abilities_Registry {
 	 * @return WP_Ability|null The registered ability instance on success, null on failure.
 	 */
 	public function register( string $name, array $args ): ?WP_Ability {
-		if ( ! preg_match( '/^[a-z0-9-]+\/[a-z0-9-]+$/', $name ) ) {
+		if ( ! preg_match( '/^[a-z0-9-]+\/([a-z0-9-]+\/)?[a-z0-9-]+$/', $name ) ) {
```

Comment:
According to the example, where the second path segment is `v1`, the regex would seem to rather be:

```suggestion
		if ( ! preg_match( '/^[a-z0-9-]+\/(v[0-9]+\/)?[a-z0-9-]+$/', $name ) ) {
```

Reply:
I thought about doing that but wasn't sure if we should have a direct opinion about how people version things. We're also not enforcing a `a.b.c` structure, either.

Reply:
Having an opinion may make it easier to in the future just return the latest version for some ability. If we don't enforce a structure our only option is to rely on string sorting which has unexpected results e.g: v3 is higher than v13. I expected because of keeping context small/relevant for LLMS having a function that just returns the last version of each ability will be something common.

I'm tempted to force an integer value for the version as it seems to be the simplest approach with obvious outcomes when sorted. We could also support something like  v[0-9]+(\.[0-9]+){,2} if we want to be more flexible and support vX.Y, vX.Y.Z, where X, Y, Z are considered integers for sorting purposes.

Reply:
You are right, and honestly I want to do that. This allows for things like [version_compare](https://www.php.net/manual/en/function.version-compare.php) being used. If someone doesn't follow that then I guess it could have odd behavior.

This does bring my [deprecation PR](https://github.com/WordPress/wordpress-develop/pull/10507) to mind as well, which _could_ refer to a version.

I feel like there are two directions we could go:
1. No strict boundaries around versioning; just let implementers do their own thing
2. Enforce PHP versioning so we can use versioning functions

Reply:
I think we should enforce a versioning system but I don't think we need the full php version system something like v10.X.Y where both X and Y are optional should be enough, consumers of abilities like the JS client package, or something consuming the rest API in a random programming language may also need to compare versions, and having the full php versioning system with RC etc adds more complexity.

Reply:
If start with a strict subset in the future we can expand, but starting broad and then stricting may not be possible that's also the reason I would prefer a stricter approach at the beginning.